pkg/tls/tls.go

```diff
@@ -52,3 +57,59 @@ func MakeCA(signerName string) (*crypto.CA, error) {
 		Config:          caConfig,
 	}, nil
 }
+
+const (
+	TigeraCASecretName          = "tigera-ca-private"
+	TrustedCertConfigMapName    = "tigera-ca-bundle"
+	TrustedCertConfigMapKeyName = "tigera-ca-bundle.crt"
+	TrustedCertVolumeMountPath  = "/etc/pki/tls/certs/"
+	TrustedCertBundleMountPath  = "/etc/pki/tls/certs/tigera-ca-bundle.crt"
+)
+
+// TigeraCA is a signer of new certificates and has methods to retrieve existing KeyPairs and Certificates. If a user
+// brings their own secrets, TigeraCA will preserve and return them.
+type TigeraCA interface {
+	// CertificateManagement returns the CertificateManagement object or nil if it is not configured.
+	CertificateManagement() *operatorv1.CertificateManagement
```

Comment:
I'm thinking out loud here, but I'm wondering if we need to expose whether or not certificate management is enabled. The other pattern we could use is to have these tls interfaces return the objects as a list of runtime objects needed for a component to use certificate management if it was enabled. That would mean also removeing the `InitContainer`, `VolumenMount`, and `Volume` functions.

There could be one function that's always called (regardless of whether certificate management is enabled or not) that takes a common list of parameters and returns the runtime objects that need to be created. This would save use having if statements in every single component to check (and also missing the proper sequence of if statements for new components) to render the appropriate objects.

Reply:
I like the idea and I have been trying to do exactly what you propose, but if you look at all the render code we have today, it just won't work out nicely. Most render files break down the different parts of specs into different funcs and although I intuitively like your suggestion, in practice it is not nice.